MetaMask/metamask-extension
# feat: Disable services with a Basic configuration toggle

## **Description**

This PR adds a Basic configuration toggle that disables the following services:
- Token API: Automatic token detection (also on new netowork info component)
- Price API: Currency rate check
- Phishing List: Phish detect

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23456?quickstart=1)

## **Related issues**

Fixes: https://app.zenhub.com/workspaces/metamask-wallet-ux-63529dce65cbb100265a3842/issues/gh/metamask/metamask-planning/2077

## **Manual testing steps**

1. Go to Settings
2. Go to Security & privacy
3. Enable Basic Configuration
4. Check that 'Autodetect tokens', 'Show balance and token price checker', 'Use phishing detection' are all turned off
5. Check that when rendering new-network-info.js the component does not make any requests to `https://token-api.metaswap.codefi.network/tokens/`

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/11148144/0c8fd575-3780-47a2-b23e-c88295db4d52


### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
